types/src/addressable_entity.rs

```diff
@@ -1715,7 +1715,16 @@ impl AddressableEntity {
 
     /// Determines if `AddressableEntity` is compatible with a given `ProtocolVersion`.
     pub fn is_compatible_protocol_version(&self, protocol_version: ProtocolVersion) -> bool {
-        self.protocol_version.value().major == protocol_version.value().major
+        let entity_protocol_version = self.protocol_version.value();
+        let context_protocol_version = protocol_version.value();
+        if entity_protocol_version.major == context_protocol_version.major {
+            return true;
+        }
+        if entity_protocol_version.major == 1 && context_protocol_version.major == 2 {
```

Comment:
it actually is absolute 1 ? 2 ... new side chains must start up from V2 given the nature of the migration / uplift logic. we discussed this in stand up iirc last week.

legacy contracts and addressable entities have a field that tracks what the protocol version was when they were installed / stored. since we are continuing to provide retrocompat support for 1.x contracts in 2.0, that means such an entity that was stored in 1.x is compatible with a 2.0 runtime instance ... and that is what this check is allowing.

thus the two currently supported cases are matching major versions, or explicitly 1.x entity 2.x runtime